src/apiron/endpoint/endpoint.py

```diff
@@ -1,25 +1,44 @@
 import logging
 import string
+import sys
 import warnings
 from functools import partial, update_wrapper
-from typing import Any, Dict, Iterable, List, Union
+from typing import Any, Callable, Dict, Iterable, List, TypeVar, Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 10):
+        from typing import Concatenate, ParamSpec
+    else:
+        from typing_extensions import Concatenate, ParamSpec
+
+    P = ParamSpec("P")
+    R = TypeVar("R")
 
 import requests
 
 from apiron import client
 from apiron.exceptions import UnfulfilledParameterException
+from apiron.service import Service
```

Comment:
Is this import necessary? It creates an import cycle which might work for now, but with the forward reference as a string below I wonder if mypy remains happy if this is removed.

```suggestion
```

Reply:
It won't remain happy, but we can move this import into the `if TYPE_CHECKING:` block. I'll do that.